spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/core/SpannerTransactionManager.java

```diff
@@ -100,6 +100,7 @@ protected void doBegin(Object transactionObject, TransactionDefinition transacti
 		if (transactionDefinition.isReadOnly()) {
 			final ReadContext targetTransactionContext = this.databaseClientProvider.get()
 					.readOnlyTransaction();
+			tx.isReadOnly = true;
```

Comment:
Don't you need to set it to `false` in the `else`, since this is a thread-local variable and will be recycled.

Reply:
Will do. I didn't know that about thread-local variables :)

Reply:
Done.